exporter/ststopologyexporter/internal/topology.go

```diff
@@ -121,35 +116,69 @@ func (c *ComponentsCollection) addHostResource(attrs *pcommon.Map, instance stri
 		}
 		c.addRelation(hostIdentifier, instance, "executes")
 	} else if faasId, ok := attrs.Get("faas.id"); ok {
-		c.functions[faasId.AsString()] = &Component{
-			fmt.Sprintf("urn:opentelemetry:function/%s", faasId.AsString()),
+		faasIdentifier := fmt.Sprintf("urn:opentelemetry:function/%s", faasId.AsString())
+		c.components[faasIdentifier] = &Component{
+			faasIdentifier,
 			ComponentType{
 				"function",
 			},
 			newComponentData().
+				withName(faasId.AsString()).
 				withEnvironment(attrs).
 				withLayer("urn:stackpack:common:layer:serverless").
 				withVersion(attrs, "faas.version").
 				withTagPrefix(attrs, "faas").
 				withTagPrefix(attrs, "cloud"),
 		}
+		c.addRelation(faasIdentifier, instance, "executes")
 	} else if taskId, ok := attrs.Get("aws.ecs.task.id"); ok {
-		c.tasks[taskId.AsString()] = &Component{
-			fmt.Sprintf("urn:opentelemetry:task/%s", taskId.AsString()),
+		taskIdentifier := fmt.Sprintf("urn:opentelemetry:task/%s", taskId.AsString())
+		c.components[taskIdentifier] = &Component{
+			taskIdentifier,
 			ComponentType{
 				"task",
 			},
 			newComponentData().
+				withName(taskId.AsString()).
 				withEnvironment(attrs).
 				withLayer("urn:stackpack:common:layer:serverless").
 				withTagPrefix(attrs, "aws.ecs").
 				withTagPrefix(attrs, "cloud"),
 		}
+		c.addRelation(taskIdentifier, instance, "executes")
 	}
 }
 
-func (c *ComponentsCollection) AddConnection(attrs *pcommon.Map) bool {
+func (c *ComponentsCollection) addKubernetesRelation(attrs *pcommon.Map, instance string) {
 	reqAttrs := make(map[string]string, 3)
+	for _, key := range []string{
+		"k8s.cluster.name",
+		"k8s.namespace.name",
+		"k8s.pod.name",
+	} {
+		value, ok := attrs.Get(key)
+		if !ok {
+			return
+		}
+		reqAttrs[key] = value.AsString()
+	}
+	podIdentifier := fmt.Sprintf("urn:opentelemetry:kubernetes:/%s:%s:pod/%s", reqAttrs["k8s.cluster.name"], reqAttrs["k8s.namespace.name"], reqAttrs["k8s.pod.name"])
+	c.components[podIdentifier] = &Component{
+		podIdentifier,
+		ComponentType{
+			"pod",
+		},
+		newComponentData().
+			withNameFromAttr(attrs, "k8s.pod.name").
+			withTagValue("cluster-name", reqAttrs["k8s.cluster.name"]).
+			withTagValue("namespace", reqAttrs["k8s.namespace.name"]).
+			withIdentifier(fmt.Sprintf("urn:kubernetes:/%s:%s:pod/%s", reqAttrs["k8s.cluster.name"], reqAttrs["k8s.namespace.name"], reqAttrs["k8s.pod.name"])),
+	}
+	c.addRelation(podIdentifier, instance, "kubernetes to otel")
```

Comment:
is there a reason there are not dashes (hyphens) in the relation type? It is the case on stackstate

Reply:
this was just to be consistent with "provided by"; the dash is added automatically when looking up the RelationType object.  Apart from that, I've no preference - adding dashes might make it more clear that this is a kind of identifier for a relation type.